### ISSUE TITLE
Handle error on YT-Synch BE if Youtube channel is suspended by the Youtube due to TOS violations

### ISSUE THREAD

**Issue by zeeshanakram3** [Joystream/youtube-synch#211]
# Context

The Youtube Channel https://www.youtube.com/channel/UCaTNgDeB8XFQjXqW_6tib_w exists however, it has been terminated and most of the information associated with it has been redacted by Youtube due to TOS violations.

<img width="1181" alt="image" src="https://github.com/Joystream/youtube-synch/assets/37098720/827e12b4-0f90-4884-8f68-4ac35005f81c">

The problem is that YT-Synch does not handle this scenario properly so if some channel is terminated on Youtube, YT-Synch BE will keep on trying to fetch channel & videos info in each polling cycle (as if the channel is valid), which would result in the following error in YT-Synch backend.

```
{
                "message": "The YouTube account of the authenticated user is suspended. In case the authenticated user is acting on behalf of another Google account, then this error refers to the latter.",
                "domain": "youtube.common",
                "reason": "authenticatedUserAccountSuspended"
            }

```

# Fix

If the channel has been suspended by Youtube, mark its state in YT-synch backend as `OutedOut` and exclude it from syncing.

**Comment by bedeho** [comment]
Very nice job sir, so this needs a new state varaition in the yt-synch backend?

**Comment by zeeshanakram3** [comment]
> so this needs a new state varaition in the yt-synch backend?

No, the `OptedOut` state already exists in the backend, however, currently, it's assigned when the channel removes YPP collaborator access or revokes the Gleev app's permission from the Google account settings page. So I am thinking of adding the same state for this scenario, since the outcome of the action is the same, i.e., the channel is no longer an active participant of YPP.
